src/main/java/practice/StreamPractice.java

```diff
@@ -50,7 +69,11 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> (p.getSex() == Person.Sex.MAN) ? (p.getAge() >= fromAge
+                        && p.getAge() <= maleToAge) : (p.getAge() >= fromAge
+                        && p.getAge() <= femaleToAge))
```

Comment:
```suggestion
                .filter(customPredicate)
```
I'm not sure, but maybe It will be better to create own predicate instead so complicated conditions?

Reply:
It's the same. This predicate has only one usage